File: mobile/lib/services/scholarship_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class ScholarshipService {
  final String apiBaseUrl;

  ScholarshipService(this.apiBaseUrl);

  Future<dynamic> allocateScholarship(String studentId, String criteria) async {
    final response = await http.post(
      Uri.parse('$apiBaseUrl/scholarships/allocate'),
      headers: {'Content-Type': 'application/json'},
      body: json.encode({'studentId': studentId, 'criteria': criteria}),
    );
    return json.decode(response.body);
  }
}

File: frontend/src/services/scholarshipService.js
import api from './api';

export const allocateScholarship = async (studentId, criteria) => {
  try {
    const response = await api.post('/scholarships/allocate', { studentId, criteria });
    return response.data;
  } catch (error) {
    console.error('Failed to allocate scholarship:', error);
    throw new Error(error.response?.data?.message || 'Failed to allocate scholarship');
  }
};
